roles/lib_openshift/src/class/oc_adm_ca_server_cert.py

```diff
@@ -0,0 +1,103 @@
+# pylint: skip-file
+
+class CAServerCertConfig(OpenShiftCLIConfig):
+    ''' CAServerCertConfig is a DTO for the oc adm ca command '''
+    def __init__(self, kubeconfig, verbose, ca_options):
+        super(CertificateAuthorityConfig, self).__init__('ca', None, kubeconfig, ca_options)
+        self.kubeconfig = kubeconfig
+        self.verbose = verbose
+        self._ca = ca_options
+
+class CAServerCert(OpenShiftCLI):
+    ''' Class to wrap the oc adm ca create-server-cert command line'''
+    def __init__(self,
+                 config,
+                 verbose=False):
+        ''' Constructor for oadm ca '''
+        super(CAServerCert, self).__init__(None, config.kubeconfig, verbose)
+        self.config = config
+        self.verbose = verbose
+
+    def get(self):
+        '''get the current cert file
+
+           If a file exists by the same name in the specified location then the cert exists
+        '''
+        cert = self.config.config_options['cert']['value']
+        if cert and os.path.exists(cert):
+            return open(cert).read()
+
+        return None
+
+    def create(self):
+        '''run openshift oc adm ca create-server-cert cmd'''
+        options = self.config.to_option_list()
+
+        cmd = ['ca', 'create-server-cert']
+        cmd.extend(options)
+
+        return self.openshift_cmd(cmd, oadm=True)
+
+    def exists(self):
+        ''' check whether the certificate exists and has the clusterIP '''
+
+        cert_path = self.config.config_options['cert']['value']
+        if not os.path.exists(cert_path):
+            return False
+
+        # Would prefer pyopenssl but is not installed.  
```

Comment:
@detiber, I made the note because I saw the emails with Tim's module.  We also hit this when trying operations with pyopenssl in the operations environment. 

We need a solution for this.  If ssl.SSLContext would work then we could look into that.  What package is that in?  Is that in the stdlib?  If so, we should investigate.